service/src/config.rs

```diff
@@ -6,45 +6,87 @@ pub struct Config {
 	pub node_ip: String,
 	pub node_port: String,
 	pub worker_ip: String,
-	pub worker_rpc_port: String,
-	/// listening port for the mutual-remote attestation requests
-	pub worker_mu_ra_port: String,
-	/// Todo: Is this deprecated? I can only see it in `enclave_perform_ra`
-	pub ext_api_url: Option<String>,
+	/// Worker address that will be advertised on the parentchain. Should be used when the worker is running
+	/// behind an nginx or docker server.
+	pub external_worker_address: Option<String>,
+	/// Port to directly communicate with the trusted tls server inside the enclave.
+	pub trusted_worker_port: String,
+	/// Port to the untrusted ws of the validateer.
+	pub untrusted_worker_port: String,
+	/// Port for mutual-remote attestation requests.
+	pub mu_ra_port: String,
 }
 
 impl Config {
 	pub fn new(
 		node_ip: String,
 		node_port: String,
 		worker_ip: String,
-		worker_rpc_port: String,
-		worker_mu_ra_port: String,
+		external_worker_address: Option<String>,
+		trusted_worker_port: String,
+		untrusted_worker_port: String,
+		mu_ra_port: String,
 	) -> Self {
 		Self {
 			node_ip,
 			node_port,
 			worker_ip,
-			worker_rpc_port,
-			worker_mu_ra_port,
-			ext_api_url: None,
+			external_worker_address,
+			trusted_worker_port,
+			untrusted_worker_port,
+			mu_ra_port,
 		}
 	}
 
+	/// Returns the client url of the node (including ws://).
 	pub fn node_url(&self) -> String {
 		format!("{}:{}", self.node_ip, self.node_port)
 	}
 
-	pub fn worker_url(&self) -> String {
-		format!("{}:{}", self.worker_ip, self.worker_rpc_port)
+	pub fn trusted_worker_url_internal(&self) -> String {
+		format!("{}:{}", self.worker_ip, self.trusted_worker_port)
+	}
+
+	/// Returns the trusted worker url that should be addressed by external clients.
+	///
+	/// In case external_address is set, it should be considered that the internal trusted server is a tls
+	/// websocket and must have have a wss:// primary to the internal worker ip.
+	pub fn trusted_worker_url_external(&self) -> String {
```

Comment:
Is this an acceptable approach @clangenb ?

Reply:
As I advised above, I think this is not acceptable as it will break reverse-proxied setups.

I suggest parsing the external address and do the following:
* if the external address contains a port, use it as is.
* if not, default to the trusted-worker-port

Reply:
All right all right ,I give up, will do 🥲

Reply:
Added this option with the function `add_port_if_necessary`. Added some unit tests as well, because config got relatively complex.